### PR TITLE
Update CMake for Kokkos 3.0

### PR DESCRIPTION
**Pull request by sslattery**
Uses new Kokkos cmake via `find_package`

### CODE REVIEW

File: unit_test/CMakeLists.txt [hunk 1]
```diff
@@ -18,7 +18,7 @@ macro(Cajita_add_tests)
   endforeach()
   set(CAJITA_UNIT_TEST_MAIN unit_test_main.cpp)
   foreach(_device ${CAJITA_SUPPORTED_DEVICES})
-    if(Kokkos_DEVICES MATCHES "${_device}")
```
Comment (dalg24) [inline unit_test/CMakeLists.txt]: Why did you change this

Comment (sslattery) [reply unit_test/CMakeLists.txt]: I was hunting for a bug and thought the issue may have been variable scope with the new CMake. I can revert back to the old version though.

**Comment by junghans** [comment]
# [Codecov](https://codecov.io/gh/ECP-copa/Cajita/pull/27?src=pr&el=h1) Report
> :exclamation: No coverage uploaded for pull request base (`master@1c3b4d5`). [Click here to learn what that means](https://docs.codecov.io/docs/error-reference#section-missing-base-commit).
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/ECP-copa/Cajita/pull/27?src=pr&el=tree)

```diff
@@            Coverage Diff            @@
##             master      #27   +/-   ##
=========================================
  Coverage          ?   83.94%           
=========================================
  Files             ?       31           
  Lines             ?     5389           
  Branches          ?        0           
=========================================
  Hits              ?     4524           
  Misses            ?      865           
  Partials          ?        0           
```

| Flag | Coverage Δ | |
|---|---|---|
| #mpicc | `83.94% <0.00%> (?)` | |


------

[Continue to review full report at Codecov](https://codecov.io/gh/ECP-copa/Cajita/pull/27?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/ECP-copa/Cajita/pull/27?src=pr&el=footer). Last update [1c3b4d5...542b730](https://codecov.io/gh/ECP-copa/Cajita/pull/27?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).
